vault/logical_system_raft.go

```diff
@@ -707,18 +707,21 @@ func (b *SystemBackend) handleStorageRaftSnapshotWrite(force bool) framework.Ope
 			b.Core.logger.Info("shutting down prior to restoring snapshot")
 			if err := b.Core.preSeal(); err != nil {
 				b.Core.logger.Error("raft snapshot restore failed preSeal", "error", err)
+				ctxCancel()
```

Comment:
Should we just defer the `ctxCancel`? :thinking:

Reply:
Yes, but: the context should only be cancelled when there is an error. Otherwise it and the cancel function will be saved in `*Core`. I'll update the code to include that.

Reply:
Fixed